src/templater.rb

```diff
@@ -119,6 +119,12 @@ def initialize(msg="Input hash may contain infinitely recursive ERB")
     def replace_erb(template, template_parameters)
       parameters_binding = template_parameters.instance_eval {binding}
       ERB.new(template).result(parameters_binding)
+    rescue NoMethodError => e
+      # May be useful to include the name of the unset parameter in this error,
+      # however this is tricky as by the time we attempt to access a method on
+      # it the unset parameter is just `nil` as far as we can see here.
```

Comment:
I did consider something like this and have a quick attempt at wrapping the fetched template parameters the other day, so we can then use the wrapper to know which unset template parameter we're attempting to access.

However the issue I ran in to is that by wrapping the return value we are in particular changing the falsiness of it; this in particular makes rendering this in some tests fail as `nested` is always truthy even if the wrapped value is `nil`: https://github.com/alces-software/metalware/blob/develop/spec/fixtures/template.erb#L6. And there seems to be no way of modifying the truthiness of Ruby objects (see https://stackoverflow.com/questions/17988923/make-a-nullobject-evaluate-to-falsy-in-ruby).

I think the alternative approach would be to track all of the unset parameters as we're accessing them so we can know when we're then attempting to access a property of them, but this then adds complexity and room for mistakes in doing this. Happy for you to take a look and see if you can improve this though if you like :slightly_smiling_face:.